test/EndToEnd/assert.ps1

```diff
@@ -231,4 +231,19 @@ function Assert-NoBindingRedirect {
                                                                                     $_.bindingRedirect.newVersion -eq $NewVersion })
 
     Assert-True ($bindings.Count -eq 0) "Binding redirect matching $Name, $OldVersion, $NewVersion found in project $($Project.Name)"
+}
+
+function Assert-StringEqual {
+    param(
+        [parameter(Mandatory = $true)]
+        $StringA,
+        [parameter(Mandatory = $true)]
+        $StringB,
+        [parameter(Mandatory = $true)]
+        $CaseSensitive
+    )
+	
+	$result = [string]::Compare($StringA, $StringB, $CaseSensitive) 
+
+	Assert-True ($result -eq 0) "Strings $StringA and $StringB are unequal witn result $result"
```

Comment:
1.  Perhaps wrap `$StringA` and `$StringB` in quotes so as to set it off from the surrounding sentence.
2.  I don't think `$result` is very useful to report.  It'll be a number, which is not easy to interpret.  I think changing line 246 to use [`String.Equals(String, String, StringComparison)`](https://msdn.microsoft.com/en-us/library/t4411bks(v=vs.110).aspx) would be simpler.
3.  Typo:  witn

Consider:

```
$comparison = if ($isCaseSensitive) { [StringComparison]::Ordinal } else { [StringComparison]::OrdinalIgnoreCase }
$areEqual = [string]::Equals($stringA, $stringB, $comparison)

Assert-True ($areEqual) "Strings `"$stringA`" and `"$stringB`" are not equal."
```